pkg/maven/settings.go

```diff
@@ -0,0 +1,150 @@
+package maven
+
+import (
+	"errors"
+	"fmt"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var getenv = os.Getenv
+
+// SettingsFileType ...
+type SettingsFileType int
+
+const (
+	// GlobalSettingsFile ...
+	GlobalSettingsFile SettingsFileType = iota
+	// ProjectSettingsFile ...
+	ProjectSettingsFile SettingsFileType = iota
+)
+
+// GetSettingsFile ...
```

Comment:
In my IDE I get the func with a yellow underline if that comment is not present alongside with a description to add some docu. Would you like to have some more reasonable docu for a method called `getSettingsFile`? I mean we can write `retrieves the settings file blah ....`. We can add more docu but in my experience speaking names a in most cases sufficient. I saw too many outdated comments (comment written at the beginning, method refactored later on without adjusting docu ...)

Reply:
Perfectly happy with your answer about the IDE, this was an honest question.

Reply:
Just a suggestion: If the need for these additional comments is a nuisance to you, you could also try and find out if there is a setting to disable this particular warning in your IDE. (VSCode does not generate such warnings for me by default.)

Reply:
I thought this would also affect all the others. I did not start with providing these comments. Yes, sure I can have a look and omit these comments. Would be interesting how more colleagues are dealing with this.

Reply:
I guess (did not look into it) that could be configured in revive: https://github.com/SAP/jenkins-library/pull/1207